api/src/Entity/Profile.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiProperty;
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\InputFilter;
+use App\Repository\ProfileRepository;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * The profile of a person using eCamp.
+ * The properties available to related eCamp users are here.
+ * Related means that they were or are collaborators in the same camp.
+ *
+ * @ORM\Entity(repositoryClass=ProfileRepository::class)
+ * @ORM\Table(name="`profile`")
+ */
+#[ApiResource(
+    collectionOperations: [
+        'get' => ['security' => 'false'],
+    ],
+    itemOperations: [
+        'get' => ['security' => 'is_fully_authenticated()'],
+        'patch' => ['security' => 'object.user === user'],
+    ],
+    denormalizationContext: ['groups' => ['write']],
+    normalizationContext: ['groups' => ['read']],
+)]
+class Profile extends BaseEntity {
+    public const EXAMPLE_EMAIL = 'bi-pi@example.com';
+    public const EXAMPLE_USERNAME = 'bipi';
+    public const EXAMPLE_FIRSTNAME = 'Robert';
+    public const EXAMPLE_SURNAME = 'Baden-Powell';
+    public const EXAMPLE_NICKNAME = 'Bi-Pi';
+    public const EXAMPLE_LANGUAGE = 'en';
+
+    /**
+     * Unique email of the user.
+     *
+     * @ORM\Column(type="string", length=64, nullable=false, unique=true)
+     */
+    #[InputFilter\Trim]
+    #[Assert\NotBlank]
+    #[Assert\Email]
+    #[ApiProperty(example: self::EXAMPLE_EMAIL)]
+    #[Groups(['read', 'write'])]
+    public ?string $email = null;
+
+    /**
+     * Unique username. Lower case alphanumeric symbols, dashes, periods and underscores only.
+     *
+     * @ORM\Column(type="string", length=32, nullable=false, unique=true)
+     */
+    #[InputFilter\Trim]
+    #[Assert\NotBlank]
+    #[Assert\Regex(pattern: '/^[a-z0-9_.-]+$/')]
+    #[ApiProperty(example: self::EXAMPLE_USERNAME)]
+    #[Groups(['read', 'create'])]
+    public ?string $username = null;
+
+    /**
+     * The user's (optional) first name.
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    #[InputFilter\Trim]
+    #[InputFilter\CleanHTML]
+    #[ApiProperty(example: self::EXAMPLE_FIRSTNAME)]
+    #[Groups(['read', 'write'])]
```

Comment:
I'd make this `#[Groups(['read'])]` only until someone implements #2357

Reply:
Habe ich jetzt einfach verschoben. Ich schlage vor wir lassen das so und implementieren #2357, das brauchen wir so oder so vor dem go live.

Reply:
Nicht sicher, ob #2357 zwingend Teil des MVPs sein muss (man kommt wahrscheinlich gut ein paar Monate ohne Wechsel von E-Mailadresse aus bzw. die wenigen Fälle könne man im Notfall auch manuell handeln). Ich würde eher für den Moment ausschalten. Wenn jemand dann #2357 vor Go-Live implementiert, perfekt. Wenn nicht, ist dann auch nicht so schlimm.

Reply:
Ok, mache ich aber in einem separaten PR.

Reply:
Ok for me